src/NuGetGallery/Controllers/AuthenticationController.cs

```
@@ -349,11 +349,23 @@ public ActionResult ChallengeAuthentication(string returnUrl, string provider)
             else
             {
                 // Gather data for view model
+                string name = null;
+                string email = null;
                 var authUI = result.Authenticator.GetUI();
-                var email = result.ExternalIdentity.GetClaimOrDefault(ClaimTypes.Email);
-                var name = result
-                    .ExternalIdentity
-                    .GetClaimOrDefault(ClaimTypes.Name);
+                try
+                {
+                    var userInfo = result.Authenticator.GetAuthInformation(result.ExternalIdentity);
+                    name = userInfo.Name;
+                    email = userInfo.Email;
+                }
+                catch (Exception)
```

Is there a more specific Exception type we expect to see? What's the scenario?
 
Should we add trace log?

---

Before this change, when we found a user trying to register a new account directly with MSA, we would show `email` we got from the auth and allow the user to edit. Going forward, we will not allow this to happen, however, for backwards compatibility(more so especially for tests) we get one or the other claim. I haven't yet modified those tests as we need to maintain backwards compatibility. Hence, for now we will swallow the exceptions. In future we will surface the error to the view model and show appropriate error to the user. No point in adding trace here yet.